### PR TITLE
Close Share Schedule & Selector modals when clicking outside

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
Close Share Schedule & Selector modals when clicking outside.

Used a new factory `outsideClickHandler` to handle the logic
Removed onBackHandler on schedule-component as no longer needed given the modal is automatically dismissed now

## Motivation and Context
Fix for https://github.com/Rise-Vision/rise-vision-apps/issues/2229

## How Has This Been Tested?
Validated all impacted modals locally and on [stage-1]

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Review by santiagonoguez — APPROVED** [review]
Looks good to me :D

File: web/scripts/schedules/directives/dtv-share-schedule-button.js [hunk 1]
```diff
@@ -37,9 +38,11 @@ angular.module('risevision.schedules.directives')
             $timeout(function () {
               if (isTooltipOpen) {
                 isTooltipOpen = false;
+                outsideClickHandler.unbind('share-schedule');
                 tooltipButton.trigger('hide');
               } else {
                 isTooltipOpen = true;
+                outsideClickHandler.bind('share-schedule', '#share-schedule-button, #share-schedule-popover', $scope.toggleTooltip);
```
Comment (alex-deaconu) [inline web/scripts/schedules/directives/dtv-share-schedule-button.js:45]: @ezequielc curious why are you adding the button `#share-schedule-button` to the selector here and in the other instances as well? Thanks.

**Review by alex-deaconu — APPROVED** [review]
@ezequielc looks good, one clarification inline. Thanks!

File: web/scripts/schedules/directives/dtv-share-schedule-button.js (hunk 1 above)
Comment (ezequielc) [reply web/scripts/schedules/directives/dtv-share-schedule-button.js:45]: @alex-deaconu Mainly for the button toggle to work as expected. By ignoring the close callback, the button toggle function will be called and close the modal. If we don't add it, the modal will be closed and reopened. Thanks

Comment (alex-deaconu) [reply web/scripts/schedules/directives/dtv-share-schedule-button.js:45]: @ezequielc sounds good, thanks for the clarification!

**Pull request merged**